common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -213,6 +213,11 @@ public boolean encryptionEnabled() {
     return conf.getBoolean("spark.network.crypto.enabled", false);
   }
 
+  /**
+   * Version number to be used by the AuthEngine key agreement protocol. Value values are 1 or 2.
+   */
+  public int authEngineVersion() { return conf.getInt("spark.network.crypto.authEngineVersion", 2); }
```

Comment:
Make the default 1, to preserve the existing behavior, and allowing backport to patch versions.
We can document to set this explicitly to to `2` to mitigate this issue.

In a later minor version, we can flip this to `2` by default.

Reply:
+1 for the above comment, too.